packages/modules/pricing/src/repositories/pricing.ts

```diff
@@ -53,28 +52,49 @@ export class PricingRepository
     }
 
     // Generate flatten key-value pairs for rule matching
-    const [ruleAttributes, priceListRuleAttributes] =
-      await this.getAttributesFromRuleTables(knex)
-
-    const allowedRuleAttributes = [
-      ...ruleAttributes,
-      ...priceListRuleAttributes,
-    ]
-
     const flattenedKeyValuePairs = flattenObjectToKeyValuePairs(context)
 
-    const flattenedContext = Object.entries(flattenedKeyValuePairs).filter(
-      ([key, value]) => {
+    // First filter by value presence
+    const candidateContext = Object.entries(flattenedKeyValuePairs).filter(
+      ([, value]) => {
         const isValuePresent = !Array.isArray(value) && isPresent(value)
         const isArrayPresent = Array.isArray(value) && value.flat(1).length
 
-        return (
-          allowedRuleAttributes.includes(key) &&
-          (isValuePresent || isArrayPresent)
-        )
+        return isValuePresent || isArrayPresent
```

Comment:
This just needs the test we spoke about regarding a full cart context, if its no longer needed, killer! 🔥